sled-agent/src/bootstrap/rss_handle.rs

```diff
@@ -0,0 +1,232 @@
+            >,
+        >(())
+    };
+
+    let log_failure = |error, _| {
+        warn!(log, "failed to start sled agent"; "error" => ?error);
+    };
+    retry_notify(internal_service_policy(), sled_agent_initialize, log_failure)
+        .await?;
+    info!(log, "Peer agent at {} initialized", url);
+    Ok(())
+}
+
+// RSS needs to send requests (and receive responses) to and from its local sled
+// agent over some communication channel. Currently RSS lives in-process with
+// sled-agent, so we can use tokio channels. If we move out-of-process we'll
+// need to switch to something like Unix domain sockets. We'll wrap the
+// communication in the types below to avoid using tokio channels directly and
+// leave a breadcrumb for where the work will need to be done to switch the
+// communication mechanism.
+fn rss_sled_agent_channel() -> (RssSledAgentSender, RssSledAgentReceiver) {
+    let (tx, rx) = mpsc::channel(32);
+    (RssSledAgentSender { inner: tx }, RssSledAgentReceiver { inner: rx })
+}
+
+type InnerInitRequest = (
+    Vec<(SocketAddrV6, SledAgentRequest)>,
+    oneshot::Sender<Result<(), String>>,
+);
+
+pub(crate) struct RssSledAgentSender {
+    inner: mpsc::Sender<InnerInitRequest>,
+}
+
+impl RssSledAgentSender {
+    pub(crate) async fn initialize_sleds(
+        &self,
+        requests: Vec<(SocketAddrV6, SledAgentRequest)>,
+    ) -> Result<(), String> {
+        let (tx, rx) = oneshot::channel();
+
+        // IPC will require real error handling, but we know the sled-agent task
+        // will not close the channel until we do and that it will always send a
+        // response, so unwrapping here is fine.
+        self.inner.send((requests, tx)).await.unwrap();
+        rx.await.unwrap()
+    }
+}
+
+struct RssSledAgentReceiver {
+    inner: mpsc::Receiver<InnerInitRequest>,
+}
+
+impl RssSledAgentReceiver {
+    async fn initialize_sleds<F, Fut>(mut self, log: &Logger, init_one_sled: F)
+    where
+        F: Fn(SocketAddrV6, SledAgentRequest) -> Fut,
+        Fut: Future<Output = Result<(), String>>,
+    {
+        let (requests, tx_response) = match self.inner.recv().await {
+            Some(requests) => requests,
+            None => {
+                warn!(
+                    log,
+                    "Failed receiving sled initialization requests from RSS",
+                );
+                return;
+            }
+        };
+
+        let mut futs = requests
+            .into_iter()
+            .map(|(addr, req)| init_one_sled(addr, req))
+            .collect::<FuturesUnordered<_>>();
+
+        // Wait for all initialization requests to complete, but stop on the
+        // first error.
+        while let Some(result) = futs.next().await {
+            if result.is_err() {
+                // IPC will require real error handling, but we know the
+                // RSS task is waiting for our response, so unwrapping here is
+                // fine (we can only panic if RSS already panicked).
```

Comment:
Can we file a bug and link it in the comment for the `unwrap` calls?

Reply:
Do you want a separate bug, or just a link to https://github.com/oxidecomputer/omicron/issues/820? Once RSS is moved to a separate process, this can't use tokio channels anymore and will have to be moved to something fundamentally different; until that happens, the `.unwrap()`s should be harmless.

Reply:
#820  seems like the right one to link! Good call

Reply:
:+1: added links in 114b4c21